File: archive/c/c-sharp/PrimeNumber.cs
if (args is not [var raw] || !ulong.TryParse(raw, out ulong number))
    return ExitWithUsage();

Console.WriteLine(IsPrime(number) ? "Prime" : "Composite");
return 0;

static bool IsPrime(ulong value)
{
    if (value < 2)
        return false;

    if (value == 2)
        return true;

    if (value % 2 == 0)
        return false;

    for (ulong divisor = 3; divisor * divisor <= value; divisor += 2)
    {
        if (value % divisor == 0)
            return false;
    }

    return true;
}

static int ExitWithUsage()
{
    Console.WriteLine("Usage: please input a non-negative integer");
    return 1;
}

File: archive/v/visual-basic/prime-number.vb
Public Module PrimeNumber

    Private Sub ShowUsage()
        Console.WriteLine("Usage: please input a non-negative integer")
        Environment.Exit(1)
    End Sub

    Private Function IsPrime(value As ULong) As Boolean

        If value <= 1UL Then Return False
        If value <> 2UL AndAlso value Mod 2UL = 0 Then Return False

        Dim limit As ULong = CULng(Math.Sqrt(value))

        Dim i As ULong = 3UL
        While i <= limit

            If value Mod i = 0UL Then
                Return False
            End If

            i += 2UL

        End While

        Return True

    End Function


    Public Function Main(args As String()) As Integer

        If args.Length <> 1 Then
            ShowUsage()
        End If

        Dim value As ULong

        If Not ULong.TryParse(args(0), value) Then
            ShowUsage()
        End If

        Console.WriteLine((If(IsPrime(value), "Prime", "Composite")))

        Return 0

    End Function

End Module